manifests/charts/istio-control/istio-discovery/files/waypoint.yaml

```diff
@@ -208,4 +208,29 @@ spec:
         {{- range .Values.global.imagePullSecrets }}
         - name: {{ . }}
         {{- end }}
-      {{- end }}
\ No newline at end of file
+      {{- end }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    {{ toJsonMap (omit .Annotations "kubectl.kubernetes.io/last-applied-configuration" "gateway.istio.io/name-override" "gateway.istio.io/service-account") | nindent 4 }}
+  labels:
+    {{ toJsonMap .Labels | nindent 4}}
+  name: {{.DeploymentName | quote}}
+  namespace: {{.Namespace | quote}}
+  ownerReferences:
+  - apiVersion: gateway.networking.k8s.io/v1beta1
+    kind: Gateway
+    name: "{{.Name}}"
+    uid: "{{.UID}}"
+spec:
+  clusterIP: None
+  ports:
+  - name: https-hbone
+    port: 15008
+    protocol: TCP
+    appProtocol: https
+  selector:
+    istio.io/gateway-name: "{{.Name}}"
```

Comment:
Is there any chance of interference with regular gateways ( the old un-namespaced references, etc) ?

Reply:
The naming is `<gateway-name>-<class name>` so its unlikely. If we merge https://github.com/istio/istio/pull/43541 its 0% chance